.github/workflows/reusable-e2e.yml

```diff
@@ -47,6 +66,8 @@ jobs:
         run: |
           cd contracts
           npm run compileSol
+          npm run hardhat &
+          sleep 5
```

Comment:
For future iterations, it might be more clear to separate `npm run hardhat` into its own step. Having it under 'Compile Contracts' can create confusion, especially for those less familiar with less familiar with our tech stack. CI scripts often serve as a how-to guide for the codebase, so clarity is key.

Reply:
Also, relying on sleep 5 to ensure the hardhat process is up may not be the most robust approach. If the process takes longer, it could lead to test failures that require manual re-triggering of CI runs. Conversely, if it's ready before 5 seconds, we're wasting time. We previously used `hnctl` for port checking to ensure the process was up and running, which might be a more reliable and efficient approach.

Reply:
oh yea sleep 5 was only there while debugging, my bad for forgetting to remove it